pkg/operator/baremetal_test.go

```diff
@@ -40,6 +41,50 @@ var (
 	expectedHttpPort                 = "6180"
 )
 
+var (
+	managed = `
+apiVersion: metal3.io/v1alpha1
+kind: Provisioning
+metadata:
+  name: test
+spec:
+  provisioningInterface: "ensp0"
+  provisioningIP: "172.30.20.3"
+  provisioningNetworkCIDR: "172.30.20.0/24"
+  provisioningDHCPExternal: false
+  provisioningDHCPRange: "172.30.20.11, 172.30.20.101"
+  provisioningOSDownloadURL: "http://172.22.0.1/images/rhcos-44.81.202001171431.0-openstack.x86_64.qcow2.gz?sha256=e98f83a2b9d4043719664a2be75fe8134dc6ca1fdbde807996622f8cc7ecd234"
+  provisioningNetwork: "managed"
+`
+	unmanaged = `
+apiVersion: metal3.io/v1alpha1
+kind: Provisioning
+metadata:
+  name: test
+spec:
+  provisioningInterface: "ensp0"
+  provisioningIP: "172.30.20.3"
+  provisioningNetworkCIDR: "172.30.20.0/24"
+  provisioningDHCPRange: ""
+  provisioningOSDownloadURL: "http://172.22.0.1/images/rhcos-44.81.202001171431.0-openstack.x86_64.qcow2.gz?sha256=e98f83a2b9d4043719664a2be75fe8134dc6ca1fdbde807996622f8cc7ecd234"
+  provisioningNetwork: "unmanaged"
+`
+	disabled = `
+apiVersion: metal3.io/v1alpha1
+kind: Provisioning
+metadata:
+  name: test
+spec:
+  provisioningInterface: "ensp0"
+  provisioningIP: "172.30.20.3"
+  provisioningNetworkCIDR: "172.30.20.0/24"
```

Comment:
@stbenjam In the disabled mode, if provisioningNetworkCIDR is not provided, we will not be able to generate the correct provisioningIP because it is expected in the format <provisiong-ip>/<CIDR>. So, my understanding is that both provisioningIP and provisioningNetworkCIDR would not be provided and only the provisioningInterface is useful in this mode.

Reply:
I think provisioningNetworkCIDR will be empty. However provisioningIP is still required, we need an IP to host the provisioning services on the external network: https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/baremetal-provisioning-optional.md#non-goals

We won't be able to calculate a default, though.

Reply:
Thanks! How do we generate provisioning IP in the format mentioned above with the CIDR? Do we expect the user to provide it with the CIDR in it?

Reply:
So, how do we calculate the provisioning IP without the networkCIDR then. Is the user required to provide it in that "generated" format? Here is where we "generate" it : https://github.com/openshift/machine-api-operator/blob/master/pkg/operator/baremetal_config.go#L90.

Reply:
That's a good point -- we have the data in the installer, since we'll know the machine network CIDR that IP is on.  We'll need a way to get it calculated there, I'm not far enough along yet to see how that part goes, hopefully next week I can start trying this with my WIP installer branch

Reply:
Ok, so I'll send you the machine network CIDR in this field, so you can use it to calculate the provisioningIP correctly. So it can work the same way it works now.

But, we should avoid passing on the provisinoingNetworkCIDR when the provisioning network is disabled to any containers if we do. I didn't take a close look at how we use that data.

Reply:
yes, agreed. To make things even more explicit, we could pass the string "external-machine-network-prefix"/<valid CIDR>. I'll leave that decision to the installer. I can make sure the implementation here is inline with what you described above.